### PR TITLE
Update TensorFlow to latest version and fix notebook for macOS

### PR DESCRIPTION
**Pull request by puddly**
The Piper sample generator currently cannot be installed on macOS due to dependency and build-time issues. I've updated the installation section of the notebook to fix these issues.

This PR also bumps TensorFlow and tf.keras to their latest release. I was originally trying to get `tensorflow-metal` working for acceleration but it only works up to 2.17.1, not 2.18.0 (both work fine with this PR). The slowest portion of training for me is actually initially generating samples with Piper, which for some reason is 20x slower on an M1 Max compared to a NVIDIA GPU.

I'll make a second PR next week with some tweaks to augmentations and for generating more negative, non-ambient samples.

### CODE REVIEW

File: notebooks/basic_training_notebook.ipynb
Comment (kahrendt) [inline notebooks/basic_training_notebook.ipynb:1]: Can you make these changes in a copied notebook file specifically marked for using on Apple silicon?

File: notebooks/basic_training_notebook.ipynb [hunk 1]
```diff
@@ -53,9 +60,7 @@
     "    !wget -O piper-sample-generator/models/en_US-libritts_r-medium.pt 'https://github.com/rhasspy/piper-sample-generator/releases/download/v2.0.0/en_US-libritts_r-medium.pt'\n",
```
Comment (kahrendt) [inline notebooks/basic_training_notebook.ipynb:60]: ``!git clone -b mps-support https://github.com/kahrendt/piper-sample-generator`` will install a version that takes advantage of Apple silicon for generating the samples faster

**Review by kahrendt — CHANGES_REQUESTED** [review]
I really appreciate your work on this, especially updating it all to work with newer versions of TensorFlow! 

Just two minor changes requested. It would be nice, at least for now, to have a second notebook Apple silicon for installing the appropriate packages and leave the current version for other platforms. Even better, if we could detect what platform the user is on and install the appropriate things from there... I don't know how involved that would be though! Would the Python ``platform`` package be enough to handle this?

With those two changes, I'll merge this into the ``november-update`` branch and then make a PR and merge it all in to the main repo. That way your future PR can target the main branch.

File: notebooks/basic_training_notebook.ipynb
Comment (puddly) [reply notebooks/basic_training_notebook.ipynb:1]: Sure! The `audio-metadata` fork is cross-platform and allows you to install Jupyter in the same virtualenv as MicroWakeWord, it's not strictly macOS-specific (and something I ran into when running things on Ubuntu).

For the other dependency, I can guard it with:

```python
if platform.system() == "Darwin":
    # `pymicro-features` is installed from a fork to support building on macOS
    !pip install 'git+https://github.com/puddly/pymicro-features@puddly/minimum-cpp-version'

# `audio-metadata` is installed from a fork to unpin `attrs` from a version that breaks Jupyter
!pip install 'git+https://github.com/whatsnowplaying/audio-metadata@d4ebb238e6a401bb1a5aaaac60c9e2b3cb30929f'
```

Do you want me to make a second notebook? Or just add a few branches to the cells that install dependencies? Other than the first two, the rest would be identical.

Comment (kahrendt) [reply notebooks/basic_training_notebook.ipynb:1]: Let's keep it in one notebook with the few branches within the cells to handle the small differences.

Comment (puddly) [reply notebooks/basic_training_notebook.ipynb:1]: Done.
